src/builder_state.rs

```diff
@@ -823,6 +823,22 @@ impl<TYPES: NodeType> BuilderProgress<TYPES> for BuilderState<TYPES> {
                     }
                 }
 
+                while let Ok(tx) = self.tx_receiver.try_recv() {
```

Comment:
1) put this before the other one.
2) get rid of the other one.
   2a) but we can leave it in for now to test.